src/helpers/authentication.ts

```diff
@@ -0,0 +1,16 @@
+const validUIAccess = (req, res, next): void => {
+  if (!req.admin && !req.user) return res.redirect('/login')
+  return next()
+}
+
+const validAdmin = (req, res, next): void => {
+  if (!req.admin) return res.status(401).send('Unauthorized')
+  return next()
+}
+
+const validUser = (req, res, next): void => {
+  if (!req.admin && !req.user) return res.status(401).send('Unauthorized')
+  return next()
+}
```

Comment:
I don't think you need these 2 functions because our code is structured in such a way these fucntions shhouldn't be used more than once. For example, you could simply put `validAdmin` inside the admin api router. and `validUser` inside the main api router. 

Main API router -> checks if user is valid, rrejects if not.
Admin API router -> (at this point, we know user exist). Check for the user's admin status.